pkg/newlog/cmd/newlogd.go

```diff
@@ -357,6 +358,11 @@ func main() {
 		case tmpLogfileInfo := <-movefileChan:
 			// handle logfile to gzip conversion work
 			doMoveCompressFile(&ps, tmpLogfileInfo)
+			// done gzip conversion, get rid of the temp log file in collect directory
+			err = os.Remove(tmpLogfileInfo.tmpfile)
+			if err != nil {
+				log.Fatal("doMoveCompressFile: remove file failed", err)
```

Comment:
Is it really needed to have this error as fatal?

Reply:
I know you are moving the code, but maybe is time to change it...

Reply:
that's a big pain in the behind... in this file alone we have 29 `Fatal`s. They are mostly related to issue that should never happen like "cannot create a temp file" or "cannot remove file" or cannot write to file. So when they crash newlogd a watchdog is going to notice and is going to restart the whole device. Will it help fixing whatever caused those issues? Idk, maybe. Ibh I'm not sure what might cause it in the first place... Like lack of free space to create a new file or write to an existing file? Then a reboot will probably fix nothing